client/service-center-client/src/main/java/org/apache/servicecomb/service/center/client/ServiceCenterClient.java

```diff
@@ -0,0 +1,381 @@
+  }
+
+  /**
+   * Get one microservice message of service-center
+   *
+   * @param serviceId
+   * @return Microservice ; when error happens,return null
+   */
+  @SuppressWarnings("unchecked")
+  public Microservice getMicroserviceByServiceId(String serviceId) {
+    try {
+      HttpResponse response = httpClient.getHttpRequest("/registry/microservices/" + serviceId, null, null);
+      if (response.getStatusCode() == HttpStatus.SC_OK) {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        JsonNode jsonNode = mapper.readTree(response.getContent());
+        return mapper.readValue(jsonNode.get("service").toString(), Microservice.class);
+      } else {
+        LOGGER.error("get service message fails, responseStatusCode={}, responseMessage={}, responseContent{}",
+            response.getStatusCode(), response.getMessage(), response.getContent());
+      }
+    } catch (IOException e) {
+      LOGGER.error("get service message fails", e);
+    }
+    return null;
+  }
+
+  /**
+   * Register microservice instances to service-center
+   *
+   * @param instance
+   * @param serviceId
+   * @return service instanceId ; when error happens,return null
+   */
+  public String registerMicroserviceInstance(MicroserviceInstance instance, String serviceId) {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      mapper.configure(SerializationFeature.WRAP_ROOT_VALUE, true);
+      HttpResponse response = httpClient.postHttpRequest("/registry/microservices/" + serviceId + "/instances", null,
+          mapper.writeValueAsString(instance));
+      if (response.getStatusCode() == HttpStatus.SC_OK) {
+        return response.getContent();
+      } else {
+        LOGGER.error("register service instance fails, responseStatusCode={}, responseMessage={}, responseContent{}",
+            response.getStatusCode(), response.getMessage(), response.getContent());
+      }
+    } catch (IOException e) {
+      LOGGER.error("register service instance fails", e);
+    }
+    return null;
+  }
+
+  /**
+   * Find microservice instances of service-center
+   *
+   * @param serviceId
+   * @return MicroserviceInstancesResponse ; when error happens,return null
+   */
+  public MicroserviceInstancesResponse getMicroserviceInstanceList(String serviceId) {
+    try {
+      HttpResponse response = httpClient
+          .getHttpRequest("/registry/microservices/" + serviceId + "/instances", null, null);
+      if (response.getStatusCode() == HttpStatus.SC_OK) {
+        ObjectMapper mapper = new ObjectMapper();
+        return mapper.readValue(response.getContent(), MicroserviceInstancesResponse.class);
+      } else {
+        LOGGER.error("get service instances list fails, responseStatusCode={}, responseMessage={}, responseContent{}",
+            response.getStatusCode(), response.getMessage(), response.getContent());
+      }
+    } catch (IOException e) {

```

Comment:
I think it would be better to throw these exceptions into the upper layer. The upper layer knows what to do next.